src/settings/plugins/keyfile/nms-keyfile-plugin.c

```diff
@@ -412,39 +429,49 @@ _sort_paths (const char **f1, const char **f2, GHashTable *paths)
 }
 
 static void
-read_connections (NMSettingsPlugin *config)
+_read_dir (GPtrArray *filenames, const char *path)
 {
-	NMSKeyfilePlugin *self = NMS_KEYFILE_PLUGIN (config);
-	NMSKeyfilePluginPrivate *priv = NMS_KEYFILE_PLUGIN_GET_PRIVATE (self);
 	GDir *dir;
-	GError *error = NULL;
 	const char *item;
-	GHashTable *alive_connections;
-	GHashTableIter iter;
-	NMSKeyfileConnection *connection;
-	GPtrArray *dead_connections = NULL;
-	guint i;
-	GPtrArray *filenames;
-	GHashTable *paths;
+	GError *error = NULL;
 
-	dir = g_dir_open (nms_keyfile_utils_get_path (), 0, &error);
+	dir = g_dir_open (path, 0, &error);
 	if (!dir) {
 		_LOGW ("cannot read directory '%s': %s",
-		             nms_keyfile_utils_get_path (),
+		             path,
 		             error->message);
 		g_clear_error (&error);
 		return;
 	}
 
-	alive_connections = g_hash_table_new (nm_direct_hash, NULL);
-
-	filenames = g_ptr_array_new_with_free_func (g_free);
 	while ((item = g_dir_read_name (dir))) {
 		if (nms_keyfile_utils_should_ignore_file (item))
 			continue;
-		g_ptr_array_add (filenames, g_build_filename (nms_keyfile_utils_get_path (), item, NULL));
+		g_ptr_array_add (filenames, g_build_filename (path, item, NULL));
```

Comment:
in /etc we accept almost all files (sans same extensions via `nms_keyfile_utils_should_ignore_file()`. I always thought that is a mistake, and we should only load files with a particular extension (and write them as such). We cannot change that for /etc, but can we do that for /run? (read and write).